### DISCUSSION TITLE
Share files between projects

### DISCUSSION THREAD

**Discussion by maidiries** [adonisjs/core#4090] [Help]
Hello there!

First of all I want to thank all the contributors of this amazing project, it's truly incredible.

My question is the following:
My project structure is like this:

/myproject
/myproject/frontend
/myproject/shared-ts-files
/myproject/adonisjs-backend

The idea of the shared-ts-files would be to store things like types for the API and maybe other shared logic. But the folder would live outside of the adonisjs project folder

What would be the config needed in tsconfig.json / .adonisrc.json to allow this?

**Comment by WeyIin** [comment]
I think I know what you mean;
At my current company we do something like that, although it is a PHP Symfony rather than a javascript / typescript / nodejs project.

Anyway what you want to look at is basically create a package and host it (either on npm or somewhere else), could be private or public, for private you'll want to set up authentication to the server.

I can't exactly tell you how to do all that in npm since I've never done it myself either, but that seem like a logical way to go. A package which can be used elsewhere.
